test/com/sun/jna/VarArgsTest.java

```diff
@@ -42,8 +42,8 @@
         }
         public int addVarArgs(String fmt, Number... args);
         public String returnStringVarArgs(String fmt, Object... args);
-        public void modifyStructureVarArgs(String fmt, Object arg1, Object... args);
-        public String returnStringVarArgs2(String... args);
+        public void modifyStructureVarArgs(String fmt, Object... args);
+        public String returnStringVarArgs2(String fmt, String... args);
```

Comment:
This needs more explanation - linux arm64 lets you happily call these methods without breaking, so this raises the question what makes mac OS so special here.

Reply:
This is what Apple's documentation says about it:
> The `x86_64` and `arm64` architectures have different calling conventions for variadic functions—functions with a variable number of parameters. On `x86_64`, the compiler treats fixed and variadic parameters the same, placing parameters in registers first and only using the stack when no more registers are available. On `arm64`, the compiler always places variadic parameters on the stack, regardless of whether registers are available. If you implement a function with fixed parameters, but redeclare it with variadic parameters, the mismatch causes unexpected behavior at runtime.

https://developer.apple.com/documentation/apple_silicon/addressing_architectural_differences_in_your_macos_code#overview

Reply:
Ah - thank you I had to read multiple times to grasp the real change.